funsor/distribution.py

```diff
@@ -142,7 +173,8 @@ def unscaled_sample(self, sampled_vars, sample_inputs, rng_key=None):
         sample_shape = tuple(v.size for v in sample_inputs.values())
         sample_args = (sample_shape,) if get_backend() == "torch" else (rng_key, sample_shape)
         if self.has_rsample:
-            raw_value = raw_dist.rsample(*sample_args)
+            # TODO fix this hack by adding rsample and has_rsample to Independent upstream in NumPyro
+            raw_value = getattr(raw_dist, "rsample", raw_dist.sample)(*sample_args)
```

Comment:
@fehiepsi it looks like `Independent` and other distribution modifiers like `MaskedDistribution` don't implement `rsample`/`has_rsample`. Is this intentional?

Reply:
Yes, currently `rsample` or `has_rsample` is not available in NumPyro (they are not used by any inference algorithms in NumPyro). In [jax backend](https://github.com/pyro-ppl/funsor/blob/master/funsor/jax/distributions.py#L114), I also added a similar workaround. Could you add this fix [here](https://github.com/pyro-ppl/funsor/blob/master/funsor/jax/distributions.py#L218)?

Reply:
Ok, thanks, I pushed this patch